Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Corefinement/Visitor.h

```diff
@@ -339,7 +339,9 @@ class Visitor{
       Node_id node_id,
       TriangleMesh& tm)
   {
-    mesh_to_vertex_to_node_id[&tm].insert(std::make_pair(target(h,tm),node_id));
+    CGAL_assertion_code(bool insert_ok = )
+    mesh_to_vertex_to_node_id[&tm].insert(std::make_pair(target(h,tm),node_id)).second;
+    CGAL_assertion(insert_ok || mesh_to_vertex_to_node_id[&tm][target(h,tm)]==node_id);
```

Comment:
With this code, there is a warning [with recent clang in Release][1]:
```
[ 50%] Building CXX object CMakeFiles/autorefinement_sm.dir/autorefinement_sm.cpp.o
/bin/clang++  -DCGAL_EIGEN3_ENABLED -DCGAL_TEST_SUITE=1 -DCGAL_USE_GMP -DCGAL_USE_GMPXX -DCGAL_USE_MPFR -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-Release/test/Polygon_mesh_processing/../../include -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-Release/test/Polygon_mesh_processing -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-Release/include -I/mnt/testsuite/include -isystem /usr/include/eigen3  -Wall -O3 -DCGAL_NDEBUG   -o CMakeFiles/autorefinement_sm.dir/autorefinement_sm.cpp.o -c /home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-Release/test/Polygon_mesh_processing/autorefinement_sm.cpp
In file included from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-Release/test/Polygon_mesh_processing/autorefinement_sm.cpp:5:
In file included from /mnt/testsuite/include/CGAL/Polygon_mesh_processing/corefinement.h:30:
/mnt/testsuite/include/CGAL/Polygon_mesh_processing/internal/Corefinement/Visitor.h:343:81: warning: expression result unused [-Wunused-value]
    mesh_to_vertex_to_node_id[&tm].insert(std::make_pair(target(h,tm),node_id)).second;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~
```
[1]: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-189/Polygon_mesh_processing/TestReport_lrineau_ArchLinux-clang-Release.gz

I do not see an easy way to workaround it. Do we need an extra macro that evaluates to `(void)` only if the assertions are disabled?

Reply:
I'll hide .second in `CGAL_assertion_code()`